viam-cartographer/CMakeLists.txt

```diff
@@ -1,237 +1,80 @@
 cmake_minimum_required(VERSION 3.2)
 
 project(viam-cartographer)
-add_definitions(-DABSL_LEGACY_THREAD_ANNOTATIONS)
+set (CMAKE_CXX_STANDARD 17)
 
-set(VIAM_CARTOGRAPHER_MAJOR_VERSION 1)
-set(VIAM_CARTOGRAPHER_MINOR_VERSION 0)
-set(VIAM_CARTOGRAPHER_PATCH_VERSION 0)
-set(VIAM_CARTOGRAPHER_VERSION ${CARTOGRAPHER_MAJOR_VERSION}.${CARTOGRAPHER_MINOR_VERSION}.${CARTOGRAPHER_PATCH_VERSION})
-set(VIAM_CARTOGRAPHER_SOVERSION ${CARTOGRAPHER_MAJOR_VERSION}.${CARTOGRAPHER_MINOR_VERSION})
+if(DEFINED ENV{API_SOURCE_DIR})
+  set(API_SOURCE_DIR $ENV{API_SOURCE_DIR})
+else()
+  set(API_SOURCE_DIR ${PROJECT_SOURCE_DIR}/..)
+endif()
+set(GRPC_SRC_DIR ${API_SOURCE_DIR}/grpc/cpp/gen)
 
-include("${PROJECT_SOURCE_DIR}/cmake/functions.cmake")
+if(NOT EXISTS ${GRPC_SRC_DIR})
+message(FATAL_ERROR
+  "RDK C++ gRPC gen files not found. Please specify the absolute path to your API_SOURCE_DIR e.g. 'export API_SOURCE_DIR=/home/user/api'"
+)
+endif()
 
-find_package(absl REQUIRED)
-find_package(Ceres REQUIRED COMPONENTS SuiteSparse)
-find_package(Lua 5.2 REQUIRED)
+message("PROJECT SOURCE DIR: " ${PROJECT_SOURCE_DIR})
+message("API SOURCE DIR: " ${API_SOURCE_DIR})
+message("GRPC SOURCE DIR: " ${GRPC_SRC_DIR})
 
-# include files/library from cartographer submodule
-include_directories(${PROJECT_SOURCE_DIR}/cartographer)
-include_directories(${PROJECT_SOURCE_DIR}/cartographer/build)
-link_libraries(${PROJECT_SOURCE_DIR}/cartographer/build/libcartographer.a)
 
-find_package(PCL REQUIRED)
-include_directories(${PCL_INCLUDE_DIRS})
-link_directories(${PCL_LIBRARY_DIRS})
-add_definitions(${PCL_DEFINITIONS})
+# dependencies
+add_subdirectory(cartographer)
 
-find_package(OpenSSL REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(GRPCPP REQUIRED grpc++)
```

Comment:
No, because we're not directly using it or the variables it exports in CMake. The inner "cartographer" target already depends on it, and we depend on it in turn, that's sufficient as far as I'm aware.